crates/dapf/src/acceptance/mod.rs

```diff
@@ -50,12 +50,92 @@ use std::{
     env,
     ops::Range,
     path::PathBuf,
+    sync::atomic::{AtomicUsize, Ordering},
     time::{Duration, Instant, SystemTime},
 };
-use tokio::sync::Barrier;
+use tokio::sync::{Barrier, Semaphore};
 use tracing::{info, instrument};
 use url::Url;
 
+/// Stride controls the rate at which requests are made during an aggregation job.
+///
+/// For example, for `len` = 10 and `wait_time` = 1s, every one second 10 requests will be made
+/// simultaneously.
+#[derive(Clone, Copy, Debug)]
+pub struct LoadControlStride {
+    pub len: usize,
+    pub wait_time: Duration,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct LoadControlParams {
+    max_concurrent_requests: Option<usize>,
+    min_requests_before_starting: usize,
+    stride: Option<LoadControlStride>,
+}
+
+impl LoadControlParams {
+    pub fn new<M, B>(
+        max_concurrent_requests: M,
+        min_requests_before_starting: usize,
+        stride: B,
+    ) -> Self
+    where
+        M: Into<Option<usize>>,
+        B: Into<Option<LoadControlStride>>,
+    {
+        Self {
+            max_concurrent_requests: max_concurrent_requests.into(),
+            min_requests_before_starting,
+            stride: stride.into(),
+        }
+    }
+
+    pub fn max_requests_before_starting(
+        reports_per_batch: usize,
+        reports_per_agg_job: usize,
+    ) -> usize {
+        reports_per_batch / reports_per_agg_job
+            + usize::from(reports_per_batch % reports_per_agg_job != 0)
```

Comment:
Suggest adding a comment explaining the math.

Reply:
done